graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+#
+# Copyright (C) 2024 Telicent Limited
+#
+openapi: 3.0.3
+info:
+  title: Graph Smart-Cache API
+  description: |
+    OpenAPI Specification for interacting with the Telicent Smart-Cache Graph Server.
+    
+    **Note:** This description is obviously configuration dependent with the endpoints below being enabled/disabled as the relevant deployment requires.
+  version: 0.82.8
+servers:
+  - url: http://localhost:3030
+    description: Local Fuseki server instance
+paths:
+  /{dataset}/query:
+    get:
+      summary: Execute a SPARQL Query
+      description: |
+        Submit a SPARQL query against a dataset. 
+        
+        **Note:** Depending on configuration there can be equivalent /get /read or /sparql endpoints with identical functionality.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+        - $ref: '#/components/parameters/sparqlQuery'
+      responses:
+        200:
+          description: Query executed successfully.
+        400:
+          description: Invalid query or bad request.
+        401:
+          description: Unauthorised query.
+        404:
+          description: Endpoint not found i.e. incorrect dataset.
+        500:
+          description: Internal Server Error i.e. the server failed to process the request.
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/update:
+    post:
+      summary: Execute a SPARQL Update
+      description: | 
+        Submit a SPARQL update against a dataset and, by means of the RDF Patches generated, also update the other Smart Caches.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        required: true
+        content:
+          application/sparql-update:
+            schema:
+              type: string
+              description: SPARQL Update query string.
+      responses:
+        200:
+          description: Update executed successfully.
+        400:
+          description: Invalid query or bad request.
+        401:
+          description: Unauthorised query.
+        404:
+          description: Endpoint not found i.e. incorrect dataset.
+        500:
+          description: Internal Server Error i.e. the server failed to process the request.
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/shacl:
+    get:
+      summary: Retrieve SHACL shapes for a dataset
+      description: |
+        Retrieve the SHACL validation shapes associated with a specific dataset.
+        The shapes are returned in RDF format and can be used to validate data against the dataset's SHACL rules.
```

Comment:
Might want @robwtelicent to clarify this here.

I thought this endpoint is only for validating data with both `GET` and `POST` requiring you to provide the shapes you want to validate with so not sure if this endpoint definition matches what Jena actually implements?

Reply:
Yeah, upon review it's only POST. 
https://github.com/apache/jena/blob/main/jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/SHACL_Validation.java

False assumption on my part. I'll remove it.

Reply:
I believe the `/shacl` endpoint only supports `POST` as shown [here](https://github.com/apache/jena/blob/6313965ac01658dc211b4234b47860ab9ce51274/jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/SHACL_Validation.java#L51). 

SHACL shapes are not necessarily loaded as part of the datatset, and if they were they could be retrieved through a SPARQL query if required. There is a argument for making SHACL shapes resolvable at their URL so that they can be easily referenced, in a similar way to how some ontologies are exposed (like https://www.w3.org/ns/shacl.ttl) but I don't think we are planning to do anything like that here.